### ISSUE TITLE
Precompute unstructured storage hashes

### ISSUE THREAD

**Issue by skozin** [lidofinance/lido-dao#206]
Depends on #181 and #204.

See https://github.com/merklejerk/solpp.

**Comment by ongrid** [comment]
Keccak is one of the cheapest SHA3 opcodes, takes just 30 gas. Using keccak() solidity function introduces the tiny difference of less than 100 gas units.
I'd suggest keeping the string identifiers in the contract to avoid fat fingering. We already used to collide statically defined hashes in tests and troubleshooting wasn't so easy. This optimization has sub-dollar impact and is not worth it IMHO.

**Comment by ongrid** [comment]
@skozin @vshvsh should we really proceed with it?

**Comment by ongrid** [comment]
1. We'll keep the hash calculations in solidity (justification is above)
2. For the string identifiers we'll follow ENS names and its native order, the same convention as in deployed.json

```
      "lido_app_lido_name": "lido.lido.eth",
      "lido_app_lidooracle_name": "lidooracle.lido.eth",
      "lido_app_node-operators-registry_name": "node-operators-registry.lido.eth",
      "lido_app_steth_name": "steth.lido.eth",
```

The first (left) part defines the variable. Uses `camelCase` notation for readability

Example:
```
bytes32 internal constant LAST_PUSH_TIME_VALUE_POSITION = keccak256("lastPushTime.lidooracle.lido.eth");
```

**Comment by skozin** [comment]
Fixed in #232.
